usr/opt/OSC200/usr/share/man/man8/XIsso.8
.\" This manpage source uses rsml coding. 
.so /usr/share/lib/tmac/sml
.so /usr/share/lib/tmac/rsml
.\"
.\"
.\" *********************************************************************
.\" *                                                                   *
.\" *      Copyright (c) Digital Equipment Corporation, 1991, 1994      *
.\" *                                                                   *
.\" *                       All Rights Reserved.                        *
.\" *                                                                   *
.\" *********************************************************************
.\"
.\"
.\" HISTORY
.\"
.\" @(#)$RCSfile: XIsso.8,v $ $Revision: 3.5.5.8 $ (DEC) $Date: 1994/01/14 23:24:16 $
.\"
.\" (c) Copyright 1990, 1991, 1992, 1993 OPEN SOFTWARE FOUNDATION, INC. 
.\" ALL RIGHTS RESERVED 
.\" OSF/1 1.2
.\"
.\" @(#)XIsso.1m	1.5 15:03:20 8/22/90 SecureWare Inc.
.\"
.\" Based on CMW+ Evaluation Version
.\"
.\" @(#)XIsso.1m	1.2 17:55:22 3/28/90 SecureWare
.\" Copyright (c) 1989,1990 SecureWare, Inc.
.\" Copyright (c) 1989,1990 SKM, L.P.
.\"build rmifdef
.\" #ifdef SEC_BASE
.\" #ifdef SEC_ROLE_X
.TH XIsso 8
.SH "NAME"
.PP
\*LXIsso\*O \- Windows interface for the information system security officer function (Enhanced Security)
.SH "SYNOPSIS"
.PP
.sS
\*L/usr/tcb/bin/XIsso\*O
.sE
.br
.ne 1.5i
.SH "DESCRIPTION"
.PP
.iX "XIsso"
The
\*LXIsso\*O
program is the interface for the person who performs the
ISSO (Information System Security Officer) functions.
It is a windowed interface that provides the capabilities to administer
the audit, accounts, and device subsystems. To use the \*LXIsso\*O program
you need to log in as root.
.\" NON_SILVER
.\" To invoke this program, you must
.\" have the \*Lisso\*O command authorization.
.PP
.\" NON_SILVER
.\" The \*LXIsso\*O
.\" program provides the sole control for the audit subsystem.
.SS "Audit"
You must start, setup, and stop auditing from the command line.
See the \*Lauditd\*O(8) and \*Lauditmask\*O(8) reference pages for details.
The
functions supported by the main Audit menu are as follows:
.PP
.VL 3m
.\" All this stuff was commented out for T1.4-4
.\" .LI "Enable"
.\" Enables the auditing process.
.\" .LI "Disable"
.\" Disables the auditing process.
.\" .LI "Collection"
.\" Lets you select the events, users, and groups to be audited.
.\" #ifdef SEC_MAC
.\" Also lets you set a process sensitivity level range.
.\" Only processes whose sensitivity levels are in this range are
.\" audited.
.\" #endif /* SEC_MAC */
.\" .LI "Maintenance"
.\" Lets you archive audit sessions to back up media, delete
.\" unwanted audit sessions, and restore previously saved audit sessions.  Audit
.\" sessions contain all the log files and data files needed to produce
.\" reports.  The Maintenance functions also let you modify audit parameters
.\" and the list of directories used by the audit subsystem.
.LI "Reporting Functions"
.VL 3m
.LI "Modify Selection Files"
This screen allows the administrator to create, modify, or delete
selection files.  Selection files contain parameters which indicate
how audit records will be selected from the raw audit trail during
report generation.  The selection parameters include things like
time interval, audit events, user id.  Any audit record matching
the selection criteria will be displayed.
.LI "Modify Deselection Files"
This screen allows the administrator to create, modify, or delete
deselection files.  A deselection file consists of tuples.  The
tuple is comprised of a host, audit ID, real UID, event, file pathname,
and access mode.  A deselection file can be used to further
reduce audit records when generating reports.  It can be used in
combination with a selection file.  Any audit record matching the
deselection criteria will be filtered out from the report stream.
.LI "Generate Reports"
This screen allows the administrator to view an audit report.
A selection file, a deselection file,  and an audit log can be selected
to generate a report.  Output options include generating a report
to a file, to a series of files sorted by audit ID, to a window on
the screen, or if audit is currently enabled, to follow the current
activity.  Report records can be in brief format or long format.  If
in brief format, the administrator can double click on the record and get
a pop-up of the long format.
.LE
.\"
.\" Lets you create and modify the selection files used to
.\" extract collected audit data.  These functions also let you create, display,
.\" and delete reports based on selected criteria applied to specific audit
.\" sessions.
.\" .LI "Statistics"
.\" Displays data that indicates audit session performance and shows you how
.\" much data is being collected.
.LE
.SS "Accounts"
.PP
The \*LXIsso\*O interface program controls the authentication
subsystem. The ISSO is generally responsible for setting the system-wide
default account parameters and modifying the account parameters of all
non-ISSO users.
.\"  Only the system administrator can create an account
.\" (see the \*LXSysAdmin\*O(8) reference page)
.\" and modify the account parameters of an ISSO account.
The functions supported by the main Accounts menu are as follows:
.VL 3m
.\" .LI "Defaults"
.\" Allows you to specify the system-wide default parameters for
.\" #ifdef SEC_MAC
.\" user clearance,
.\" #endif /* SEC_MAC */
.\" command authorizations, kernel authorizations,
.\" #ifdef SEC_PRIV
.\" base privileges,
.\" #endif /* SEC_PRIV */
.\" login control parameters, and password parameters.
.LI "Modify User Account"
Allows you to select a non-ISSO account and to modify the account
parameters. These parameters override any system-wide defaults that
you have specified. You can select
.\"  the audit events that are
.\" audited for the user,
the groups that the user can belong to, login control parameters,
audit events, and password parameters.
.LI "Modify User Account Template"
Allows you to modify any user account template.
Account templates may be created to specify account attributes for many user 
accounts. The system default template, SYS_DFLT, is always present.
.\" .LI "Lock Account"
.\" Allows you to explicitly lock any user account, including an ISSO
.\" account.
.\" .LI "Unlock Account"
.\" Allows you to explicitly unlock any user account, including an ISSO
.\" account. Both explicit locks (set from Lock Account), and
.\" implicit locks (such as user exceeded maximum number of password
.\" guesses) are cleared.
.\" #ifdef SEC_MAC
.\" .LI "Single User Sensitivity Level"
.\" Allows you to set the sensitivity level of the single-user shell.
.\" #endif /* SEC_MAC */
.LE
.SS "Devices"
The \*LXIsso\*O
interface program provides control over the device assignment
database and the terminal control database.
The ISSO is generally responsible for setting the system-wide
default device parameters and modifying the device and terminal
parameters of all devices.
.PP
The functions supported are as follows:
.VL 3m
.LI "Modify/Create Device"
Allows you to select a device and to create or modify the device
parameters. These parameters override any system-wide defaults that
you have specified.
Currently
.\"  Printers, removable devices, and
terminals can be added to your secure configuration.
(Printers and removable devices can be added or modified with \*LXIsso\*O 
interface, but the operating system only deals with terminals.)
.LI "Defaults"
Allows you to specify the system-wide default device control parameters.
.\" .LI "Device Selection"
.\" Allows you to add a printer, a removable device, or a terminal
.\" to your secure configuration.
.\" #ifdef SEC_MAC
.\" You can select whether the device is single-level or multilevel
.\" with respect to
.\" sensitivity levels.
.\" #endif /* SEC_MAC */
.\" You can specify the device
.\" pathnames that correspond to the device name.
.\" #ifdef SEC_MAC
.\" If the device is a
.\" multilevel device, you can set the sensitivity level range; for a
.\" single-level device you can set the sensitivity level.
.\" #endif /* SEC_MAC */
.\" For a terminal, you can also specify various
.\" terminal control parameters, such as number of unsuccessful logins.
.\" .LI "Lock Terminal"
.\" Allows you to explicitly lock any terminal. Once locked, a terminal
.\" cannot be used to log in to the system.
.\" .LI "Unlock Terminal"
.\" Allows you to explicitly unlock any terminal.
.\" Both explicit locks (set from Lock Terminal), and
.\" implicit locks (such as when
.\" the terminal exceeded the maximum number of login attempts) are cleared.
.\" .LI "Remove Device"
.\" Allows you to remove a device from your secure configuration.
.LE
.ne 1.5i
.SH "Resources"
.VL 3m
.LI "\*L*auditMaxMemoryPages\*O"
This resource specifies the maximum number of 256-Kilobyte pages that are
allocated by \*LXIsso\*O when receiving report data from the \*Laudit_tool\*O
command.
Once
this threshold is reached, \*LXIsso\*O discards the oldest data page to make
room for new data.  The default value is 20.  This means that up to 5 Megabytes
of report data is accessible at any one time while viewing a report.
If the Brief Report mode is selected, less data is available since \*LXIsso\*O
stores both the brief and the full records to expand brief
format into long format.
This memory is freed after each report is finished,
so it is not a cumulative amount.
.LE
.ne 1.5i
.SH "FILES"
.VL 3m
.LI "\*L/usr/tcb/bin/XIsso\*O"
Specifies the command path.
.LI "\*L/usr/lib/X11/app-defaults/XIsso\*O"
Resource file.
.LI "\*L/etc/auth/system/default\*O"
System Default database.
.LI "\*L/etc/auth/system/ttys\*O"
Terminal Control database.
.LI "\*L/etc/auth/system/devassign\*O"
Device Assignment database.
.LI "\*L/tcb/files/auth/[a-z]/*\*O"
Protected Password database.
.LI "\*L/usr/lib/X11/help/xif.decw_book\*O"
Bookreader help file.
.LI "\*L/etc/sec/site_events
Site-specific audit events.
.LI "\*L/etc/sec/event_aliases
Audit event alias specification file.
.LI "\*L/var/tcb/audit/selection\*O"
Directory containing the audit selection files.
.LI "\*L/var/tcb/audit/deselection\*O"
Directory containing the audit deselection files.
.LI "\*L/var/tcb/audit/base_events\*O"
Base system audit events.
.LE
.ne 1.5i
.SH "RELATED INFORMATION"
.PP
Commands: \*Lauditd\*O(8),
\*Laudgen\*O(8),
\*Laudit_setup\*O(8),
\*Laudit_tool\*O(8),
\*Lauditmask\*O(8),
\*Lsecsetup\*O(8),
\*LXSysAdmin\*O(8)
.PP
\*EEnhanced Security\*O
.\" #endif /* SEC_ROLE_X */
.\" #endif /* SEC_BASE */
.EQ
delim off
.EN


usr/opt/OSC200/src/seccmd/xif/isso/XIsso.c
/*
 * *****************************************************************
 * *                                                               *
 * *    Copyright (c) Digital Equipment Corporation, 1991, 1994    *
 * *                                                               *
 * *   All Rights Reserved.  Unpublished rights  reserved  under   *
 * *   the copyright laws of the United States.                    *
 * *                                                               *
 * *   The software contained on this media  is  proprietary  to   *
 * *   and  embodies  the  confidential  technology  of  Digital   *
 * *   Equipment Corporation.  Possession, use,  duplication  or   *
 * *   dissemination of the software and media is authorized only  *
 * *   pursuant to a valid written license from Digital Equipment  *
 * *   Corporation.                                                *
 * *                                                               *
 * *   RESTRICTED RIGHTS LEGEND   Use, duplication, or disclosure  *
 * *   by the U.S. Government is subject to restrictions  as  set  *
 * *   forth in Subparagraph (c)(1)(ii)  of  DFARS  252.227-7013,  *
 * *   or  in  FAR 52.227-19, as applicable.                       *
 * *                                                               *
 * *****************************************************************
 */
/*
 * HISTORY
 */
#ifndef lint
static char	*sccsid = "@(#)$RCSfile: XIsso.c,v $ $Revision: 4.2 $ (DEC) $Date: 1991/09/20 00:11:00 $";
#endif 
/*
 */
/*
 * (c) Copyright 1990, OPEN SOFTWARE FOUNDATION, INC.
 * ALL RIGHTS RESERVED
 */
/*
 * OSF/1 Release 1.0
 */

#if SEC_BASE

/*
	filename:
	    XIsso.c
	
	copyright:
		Copyright (c) 1989-1990 SecureWare Inc.
		ALL RIGHTS RESERVED

	function:
	    provide X user interface for role programs
	    
	notes:
	    This file is similar between isso, sysadmin and operator.

*/
	    
/* Common C include files */
#include <sys/types.h>
#include <sys/security.h>

/* Security includes */
#include <prot.h>

/* X Include files */
#include <X11/Intrinsic.h>
#include <Xm/Xm.h>
#include <Xm/Form.h>
#include <Xm/MenuShell.h>
#include <Xm/RowColumn.h>
#include <Xm/Separator.h>

/* Include files */
#define MAIN
#include "XMain.h"
#include "XAccounts.h"
#include "XAudit.h"
#include "XDevices.h"

/* Definitions */
#define MESSAGE_FILE	"/usr/share/lib/sechelp/XIsso/isso.msg"

/* External routines */
extern void 
	InitializeMessageHandling(), 	/* Initializes message stuff */
#if SEC_WIND
	MakeProgramTrusted(),		/* "Trusted Path Active" routine */
#endif
	LoadMessage();			/* Loads message buffer */

extern void 
	StartAccountsFunctionalUnits(),
	StopAccountsFunctionalUnits(),
	StartAuditFunctionalUnits(),
	StopAuditFunctionalUnits(),
	StartDevicesFunctionalUnits(),
	StopDevicesFunctionalUnits();

extern void 
	FileMenuBarInitialize(),
	AccountsMenuBarInitialize(),
	AuditMenuBarInitialize(),
	DevicesMenuBarInitialize(),
	HelpMenuBarInitialize();

/* Local routines */
static void StartFunctionalUnits();
void StopFunctionalUnits();

/* Local variables */
static char 
	    **msg_error_1,
	    **msg_error_2,
	    **msg_main,
	    *msg_error_1_text,
	    *msg_error_2_text,
	    *msg_main_text;
	
main(argc, argv)
	int     argc;
	char    **argv;
{
	Cardinal    n;
	Arg         args[20];
	Widget      w;
	int         i,
		    ret,
	            shell_height,
	            shell_width;
	Display     *display;

	/* Initialize security */
	set_auth_parameters(argc, argv);
	initprivs();
	forcepriv(SEC_REMOTE); /* This is not really needed */
	forcepriv(SEC_LIMIT);
	forcepriv(SEC_OWNER);
	forcepriv(SEC_CHOWN);
	forcepriv(SEC_EXECSUID);
	forcepriv(SEC_SUSPEND_AUDIT);
	forcepriv(SEC_WRITE_AUDIT);
	forcepriv(SEC_CONFIG_AUDIT);
#if SEC_MAC
	forcepriv(SEC_ALLOWMACACCESS);
#endif /* SEC_MAC */
	forcepriv(SEC_ALLOWDACACCESS);
#if SEC_ILB
	forcepriv(SEC_ALLOWILBACCESS);
	forcepriv(SEC_ILNOFLOAT);
#endif /* SEC_ILB */

	/* For utilities say who we are */
	role_program = ISSO;

	/* Get this application's messages loaded and ready */        
	InitializeMessageHandling(MESSAGE_FILE);

	/* Note that InitializeMessageHandling() will trap many
	*  problems with permissions -- the message file will not be
	*  found!  Code in this function attempts to notify the
	*  user as to problem -- error message goes to stdout
	*/
	LoadMessage("msg_main", &msg_main, &msg_main_text);
	
	/* Crank up X windows early so bulk of startup error messages can be 
	*  routed through X 
	*/
	
	/* Create application shell */
	main_shell = XtInitialize(
	    "Main",                         /* name of toplevel shell widget */
	    "XIsso",                        /* application class */
	    NULL, NULL, &argc, argv);       /* command line parameters */

	display = XtDisplay(main_shell);
	XSync(display, FALSE);

	no_form_present = FALSE;	/* Used if error before form present */

	/* Look for special command line parameter of "-trace" & "-memory" */
	save_memory = FALSE;
	for (i = 0; i < argc; i++) {
	    if (! stricmp(argv[i], msg_main[0]))
	        save_memory = TRUE;
	}

	/* Set title for window manager */
	n = 0;
	XtSetArg(args[n], XmNmappedWhenManaged,               FALSE); n++;
	XtSetArg(args[n], XmNallowShellResize,                 TRUE); n++;
	XtSetArg(args[n], XmNtitle,                     msg_main[1]); n++;
	XtSetValues(main_shell, args, n);

	/**********************************************************************/
	/* Primary application form widget                                    */
	/**********************************************************************/
	n = 0;
	XtSetArg(args[n], XmNdefaultPosition,                   TRUE); n++;
	XtSetArg(args[n], XmNresizePolicy,             XmRESIZE_NONE); n++;
	XtSetArg(args[n], XmNautoUnmanage,                     FALSE); n++;
	XtSetArg(args[n], XmNdialogStyle,         XmDIALOG_WORK_AREA); n++;
	XtSetArg(args[n], XmNtopOffset,                            0); n++;
	XtSetArg(args[n], XmNleftOffset,                           0); n++;
	XtSetArg(args[n], XmNbottomOffset,                         0); n++;
	XtSetArg(args[n], XmNrightOffset,                          0); n++;
	XtSetArg(args[n], XmNverticalSpacing,                      0); n++;
	XtSetArg(args[n], XmNhorizontalSpacing,                    0); n++;
	main_form = XmCreateForm(main_shell, "Main", args, n);
	XtManageChild(main_form);

	/**********************************************************************/
	/* Menu Bar                                                           */
	/**********************************************************************/
	/* Create menu bar for application */
	n = 0;
	XtSetArg(args[n], XmNtopAttachment,                 XmATTACH_FORM); n++;
	XtSetArg(args[n], XmNtopOffset,                                 0); n++;
	XtSetArg(args[n], XmNleftAttachment,                XmATTACH_FORM); n++;
	XtSetArg(args[n], XmNleftOffset,                                0); n++;
	XtSetArg(args[n], XmNborderWidth,                               0); n++;
	XtSetArg(args[n], XmNshadowThickness,                           0); n++;
	main_menubar = XmCreateMenuBar(main_form, "Menubar", 
	                            args, n);
	XtManageChild(main_menubar);

	/**********************************************************************/
	/* Separator bar                                                      */
	/**********************************************************************/
	n = 0;
	XtSetArg(args[n], XmNtopAttachment,               XmATTACH_WIDGET); n++;
	XtSetArg(args[n], XmNtopWidget,                      main_menubar); n++;
	XtSetArg(args[n], XmNtopOffset,                                 0); n++;
	XtSetArg(args[n], XmNleftAttachment,                XmATTACH_FORM); n++;
	XtSetArg(args[n], XmNleftOffset,                                0); n++;
	XtSetArg(args[n], XmNrightAttachment,               XmATTACH_FORM); n++;
	XtSetArg(args[n], XmNrightOffset,                               0); n++;
	XtSetArg(args[n], XmNmargin,                                    0); n++;
	XtSetArg(args[n], XmNorientation,                    XmHORIZONTAL); n++;
	XtSetArg(args[n], XmNseparatorType,                 XmDOUBLE_LINE); n++;
	w = XmCreateSeparator(main_form, "Separator", args, n);
	XtManageChild(w);
	
	/**********************************************************************/
	/* Primary attachment (i.e. mother) widget for all functions          */
	/**********************************************************************/
	n = 0;
	XtSetArg(args[n], XmNtopAttachment,           XmATTACH_WIDGET); n++;
	XtSetArg(args[n], XmNtopWidget,                             w); n++;
	XtSetArg(args[n], XmNtopOffset,                             0); n++;
	XtSetArg(args[n], XmNleftAttachment,            XmATTACH_FORM); n++;
	XtSetArg(args[n], XmNleftOffset,                            0); n++;
	XtSetArg(args[n], XmNrightAttachment,           XmATTACH_FORM); n++;
	XtSetArg(args[n], XmNrightOffset,                           0); n++;
	XtSetArg(args[n], XmNbottomAttachment,          XmATTACH_FORM); n++;
	XtSetArg(args[n], XmNbottomOffset,                          0); n++;
	XtSetArg(args[n], XmNrubberPositioning,                  True); n++;
	XtSetArg(args[n], XmNresizePolicy,              XmRESIZE_NONE); n++;
	XtSetArg(args[n], XmNautoUnmanage,                      FALSE); n++;
	XtSetArg(args[n], XmNborderWidth,                           0); n++;
	XtSetArg(args[n], XmNdialogStyle,          XmDIALOG_WORK_AREA); n++;
	XtSetArg(args[n], XmNverticalSpacing,                       0); n++;
	XtSetArg(args[n], XmNhorizontalSpacing,                     0); n++;
	mother_form = XmCreateForm(main_form, "Mother", args, n);
	XtManageChild(mother_form);

	/**********************************************************************/
	/* Initialize functional units                                        */
	/**********************************************************************/
	StartFunctionalUnits();
	XSync(display, FALSE);

	/**********************************************************************/
	/* Realize the widget (it is not mapped). Make it trusted then map it */
	/**********************************************************************/
	XtRealizeWidget(main_shell);
#if SEC_WIND
	XSync(display, FALSE);
	MakeProgramTrusted("isso");
	XSync(display, FALSE);
#endif
	XtMapWidget(main_shell);

	/**********************************************************************/
	/* Error checking                                                     */
	/**********************************************************************/
	if (! authorized_user("isso") ) {
		LoadMessage("msg_not_authorized", &msg_error_1, 
			&msg_error_1_text);
		SystemErrorMessageOpen(-1, msg_error_1, 0, NULL);
		XtMainLoop(); /* Dies */
	}
#ifdef OLD_CODE
    /* if you are running one of the setgid programs and your
     * effective user id is not equal to your real user id, you
     * need to be checked against the protected subsystems
     * database.
     */
    if (getgid() != getegid()) {
        grp = getgrgid(getegid());
        if (! authorized_user(grp->gr_name)) {
		LoadMessage("msg_not_authorized", &msg_error_1, 
			&msg_error_1_text);
            SystemErrorMessageOpen(1010, msg_error_1, 0, NULL);
            /* Dies */
	}
    }

/* 
    Check if the program is running as user "audit", group "audit" 
    and has SelfAudit privilege.  
    
    If all conditions pass, return 1, otherwise return 0.
    
    Note that this code must be run AFTER windows cranked up
*/

    /* check that program is running under proper permissions. */
    pwd = getpwnam("audit");
    grp = getgrnam("audit");
    if ( pwd == (struct passwd *) 0 ||
         geteuid() != pwd->pw_uid   ||
         grp == (struct group *) 0  ||
         getegid() != grp->gr_gid
    ) {
		LoadMessage("msg_not_authorized", &msg_error_1, 
			&msg_error_1_text);
            SystemErrorMessageOpen(1010, msg_error_1, 0, NULL);
        /* can't AUDIT incorrect user running the program */
        /* Dies */
    }
    /* check for audit audit privilege */
    getpriv(SEC_EFFECTIVE_PRIV, privmask);
    if (! ISBITSET(privmask, SEC_CONFIG_AUDIT)) {
		LoadMessage("msg_not_authorized", &msg_error_1, 
			&msg_error_1_text);
            SystemErrorMessageOpen(1010, msg_error_1, 0, NULL);
        /* Dies */
    }

#endif

#if SEC_MAC
	ret = mand_init();
	if (ret != 0) {
		LoadMessage("msg_cant_initialize_mand", &msg_error_2, 
			&msg_error_2_text);
		SystemErrorMessageOpen(-1, msg_error_2, 0, NULL);
		XtMainLoop(); /* Dies */
	}
#endif

	/**********************************************************************/
	/* Initialize the menu bar.                                           */
	/**********************************************************************/
	FileMenuBarInitialize();
	AuditMenuBarInitialize();
	AccountsMenuBarInitialize();
	DevicesMenuBarInitialize();
	HelpMenuBarInitialize("isso,XIsso.hlp");

	/* Normal start */
	XtMainLoop();
}

static void StartFunctionalUnits() 
{
	ConfirmationStart();
	ErrorMessageStart();
	StartAccountsFunctionalUnits();
	StartAuditFunctionalUnits();
	StartDevicesFunctionalUnits();
}

void StopFunctionalUnits() 
{
	ConfirmationStop();
	ErrorMessageStop();
	StopAccountsFunctionalUnits();
	StopAuditFunctionalUnits();
	StopDevicesFunctionalUnits();
}

#endif /* SecureWare */
